### DISCUSSION TITLE
"Dismiss stale pull request approvals when new commits are pushed" stopped working

### DISCUSSION THREAD

**Discussion by mkow** [community/community#54493] [Pull Requests]
### Select Topic Area

Bug

### Body

It seems that a few days ago "Dismiss stale pull request approvals when new commits are pushed" option in branch protection rules stopped to work. I approved a PR, then someone rebased it and changed the commit message, but my approval wasn't dismissed. It happened already in a few PRs this week.

I can't find anything about such a change in [GitHub's changelog](https://github.blog/changelog/), so I assume it's not intentional?

**Comment by ebensing** [comment]
Seeing this same issue. Is there any update on this?

**Comment by ashishgupta199** [comment]
Seeing this same issue. Is there any update on this?

**Comment by s2oy** [comment]
Seeing this same issue. Is there any update on this?

**Comment by njsaunders** [comment]
Had the same issue. It appears that rebasing doesn't count as a commit, which makes sense - Rebasing doesn't logically change the outcome of a merge to the target branch, so wouldn't expect it to dismiss PRs unless there was a conflict that required a code change.

**Reply by mkow** [reply]
It does, but indirectly - imagine you have a commit which renames variable X to Y and then the author rebases this PR to a new `master`. In the meantime there might have been new usages of X added. The rebase will go without conflicts and the reviews won't get dismissed, so the PR will be mergable while it shouldn't (modulo CI, but it may not find these issues).

Anyways, my bigger problem here is that the reviews are not dismissed anymore if the commit message changes, which is rather bad - I want to review commit messages and ensure they make sense before merging, because PR authors from outside of our projects often make errors there when squashing/rebasing.

**Comment by mkow** [comment]
Are there any updates on this? The bug is still present and happened e.g. on https://github.com/gramineproject/gramine/pull/2031 - the author _added a new commit_ when rebasing, and everyone's approvals were kept, so someone could have accidentally merged that without any review.
And yes, we still have "Dismiss stale pull request approvals when new commits are pushed" option set.
